lib/logstash/inputs/snmp.rb

```diff
@@ -210,9 +210,12 @@ def run(queue)
 
       Stud.stoppable_sleep(@interval) { stop? }
```

Comment:
it seems we call the stop function here? and this is empty? out of curiosity, why do we do that?

Reply:
This is a construct in Ruby where where pass code block (here `{ stop? }`) to a method. `stoppable_sleep` does the equivalent of a plain `sleep` but will execute the given code block every second to figure if there is a stop condition by returning `true` from the code block and if/when that happen `stoppable_sleep` will short circuit the ongoing sleep and return immediately. here the `stop?` method is a plugin method which returns `true` when the plugin is being stopped. 

We use that in plugins that loops at regular interval with potential long sleeps between cycles to avoid hanging the pipeline reload or shutdown process.